centreon/src/Core/Module/Infrastructure/ModuleVersionChecker.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * Copyright 2005 - 2025 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Module\Infrastructure;
+
+use Core\Module\Application\Repository\ModuleInformationRepositoryInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+final readonly class ModuleVersionChecker
+{
+    public function __construct(
+        #[Autowire(param: 'kernel.project_dir')] private string $projectDir,
+        private ModuleInformationRepositoryInterface $repository
+    ) {
+    }
+
+    public function hasANewVersionAvailable(string $moduleName): bool
+    {
+        $moduleInformation = $this->repository->findByName($moduleName);
+        if (! $moduleInformation) {
+            throw new \RuntimeException($moduleName . " is not installed");
+        }
```

Comment:
If we decided to rename the class & method accordingly to my previous comment, not finding the module could just return `false` instead of an exception.

On the other hand if a module name does not exist AT ALL, then maybe an exception could be raised.

Reply:
How do we define a module does not exists at all ? with a whitelist of modules names ? (but we don't want to make mentions explicitly of modules so this list could not be in this repo)